R/importance.R

```
@@ -36,131 +39,243 @@
   # importance results ordered by decreasing importance
-  if (importance_metric == "ratio") {
-    res <- data.table(X = names(res_list), risk_ratio = unlist(res_list))
-    return(res[order(-res$risk_ratio)])
-  } else if (importance_metric == "difference") {
-    res <- data.table(X = names(res_list), risk_difference = unlist(res_list))
-    return(res[order(-res$risk_difference)])
+  result <- data.table(covariate = names(X), metric = unlist(res_list))
+  if (!is.null(covariate_groups)) {
+    colnames(result)[1] <- "covariate_group"
+  }
+  result <- result[order(-result$metric), ]
+
+  # name the importance metric appropriately
+  metric_name <- paste0("risk_", importance_metric)
+  if (!is.null(attr(original_eval, "name"))) {
+    metric_name <- gsub("risk", attr(original_eval, "name"), metric_name)
   }
+  colnames(result)[2] <- metric_name
+
+  if (type == "permute") {
+    result <- list("result" = result, "RNGseeds" = RNGseeds)
+  }
+
+  return(result)
 }
 
 #' Variable Importance Plot
 #'
-#' @param x The 2-column \code{data.table} returned by the call to
-#'  \code{importance}, where the first column is the predictors and the second
-#'  col
+#' @param x The 2-column \code{data.table} returned by \code{importance}, where
+#'  the first column is the covariate/groups and the second column is the
+#'  importance metric.
 #' @param nvar The maximum number of predictors to be plotted. Defaults to 30.
-#' @param ... Other parameters passed to \code{\link[graphics]{dotchart}}.
 #'
 #' @return A \code{\link[graphics]{dotchart}} of variable importance.
 #'
-#' @importFrom graphics dotchart
+#' @importFrom ggplot2 ggplot geom_point coord_flip scale_x_discrete labs
+#' @importFrom data.table data.table
 #'
 #' @name importance_plot
 #' @keywords variable importance
 #'
 #' @export
-importance_plot <- function(x, nvar = min(30, nrow(x)), ...) {
+importance_plot <- function(x, nvar = min(30, nrow(x))) {
```

add `x_sorted` and `score` to a `utils::globalVariables(...)` call to suppress this error, https://github.com/tlverse/sl3/pull/350/checks?check_run_id=3655457878#step:8:71